src/features/editable-table/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { localStorageController } from '@/shared/utils';
+import { VariablesData } from '@/types/types';
+import { Button, Empty, Flex, Form, Popconfirm, Space, Table, TableProps, Typography } from 'antd';
+import { useEffect, useState } from 'react';
+import EditableCell from './editable-cell';
+import TableOperation from './operation';
+import { useTranslations } from 'next-intl';
+const { Text } = Typography;
+
+export default function EditableTable() {
+  const t = useTranslations('Variables');
+  const [form] = Form.useForm();
+  const [data, setData] = useState<VariablesData[]>([]);
+  const [isAddItem, setIsAddItem] = useState(false);
+  const [editingKey, setEditingKey] = useState<number>(0);
+
+  const editItem = async (record: Partial<VariablesData> & { key: number }) => {
+    form.setFieldsValue(record);
+    setEditingKey(record.key);
+  };
+
+  const saveItem = async (key: React.Key) => {
+    const value = (await form.validateFields()) as VariablesData;
+    setData((prev) => prev.map((item) => (item.key === key ? { ...item, ...value } : item)));
+    setEditingKey(0);
+    setIsAddItem(false);
+  };
+
+  useEffect(() => {
+    const stored = localStorageController.get('rest-variables');
+    setData(stored || []);
+  }, []);
+
+  useEffect(() => {
+    if (data.length > 0) {
+      const lastItem = data[data.length - 1];
+      const { variable, value } = lastItem;
+      const isNotEmpty = variable !== '' && value !== '';
+      if (isNotEmpty) {
+        localStorageController.set('rest-variables', data);
+      }
+    }
+  }, [data]);
+
+  const deleteItem = (key: number) => {
+    const newData = data.filter((item) => item.key !== key);
+    setData(newData);
+
+    if (editingKey == key) {
+      setIsAddItem(false);
+      setEditingKey(0);
+    }
+
+    if (newData.length === 0) {
+      localStorage.removeItem('rest-variables');
```

Comment:
```suggestion
      localStorageController.remove('rest-variables');
```

We have separate method in the localStorageController util